snuba_sdk/mql/mql.py

```diff
@@ -37,7 +37,7 @@ class InvalidMQLQueryError(Exception):
 coefficient = number / quoted_string / filter
 
 number = ~r"[0-9]+" ("." ~r"[0-9]+")?
-filter = target open_brace? (_ filter_expr _ )? close_brace? (group_by)?
+filter = target (open_brace (_ filter_expr _ )? close_brace)? (group_by)?
```

Comment:
This definition was wrong in the first place since it implies that filters can exist even without being surrounded by `{}` making the grammar ambiguous.